mmv1/products/dialogflow/api.yaml

```diff
@@ -318,4 +318,74 @@ objects:
+  - !ruby/object:Api::Resource
+    name: 'Fulfillment'
+    base_url: "projects/{{project}}/agent/fulfillment/"
+    self_link: "{{name}}"
+    update_verb: :PATCH
+    delete_verb: :PATCH
+    create_verb: :PATCH
+    create_url: projects/{{project}}/agent/fulfillment/?updateMask=name,displayName,enabled,genericWebService,features
+    update_url: projects/{{project}}/agent/fulfillment/
+    delete_url: projects/{{project}}/agent/fulfillment/?updateMask=name,displayName,enabled,genericWebService,features
+    update_mask: true
+    description: |
+      By default, your agent responds to a matched intent with a static response. If you're using one of the integration options, you can provide a more dynamic response by using fulfillment. When you enable fulfillment for an intent, Dialogflow responds to that intent by calling a service that you define. For example, if an end-user wants to schedule a haircut on Friday, your service can check your database and respond to the end-user with availability information for Friday.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/dialogflow/es/docs/fulfillment-overview'
+      api: 'https://cloud.google.com/dialogflow/es/docs/reference/rest/v2/projects.agent/getFulfillment'
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        output: true
+        description: |
+          The unique identifier of the fulfillment. 
+          Format: projects/<Project ID>/agent/fulfillment - projects/<Project ID>/locations/<Location ID>/agent/fulfillment
+      - !ruby/object:Api::Type::String
+        name: 'displayName'
+        required: true
+        description: |
+          The human-readable name of the fulfillment, unique within the agent.
+      - !ruby/object:Api::Type::Boolean
+        name: 'enabled'
+        description: |
+          Whether fulfillment is enabled.
+      - !ruby/object:Api::Type::Array
+        name: 'features'
+        description: |
+         The field defines whether the fulfillment is enabled for certain features.
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+          - !ruby/object:Api::Type::Enum
+            name: 'type'
+            description: |
+              The type of the feature that enabled for fulfillment.
+              * TYPE_UNSPECIFIED: Feature type not specified.
+              * SMALLTALK: Fulfillment is enabled for SmallTalk.
+            values:
+              - :TYPE_UNSPECIFIED
+              - :SMALLTALK
+      - !ruby/object:Api::Type::NestedObject
+        name: 'genericWebService'
+        description: |
+          Represents configuration for a generic web service. Dialogflow supports two mechanisms for authentications: - Basic authentication with username and password. - Authentication with additional authentication headers.
+        properties:
```

Comment:
I think all these subfields can be made required?

Reply:
Hi @rileykarson checking documentation and only uri field is required so I am marking it as required https://cloud.google.com/dialogflow/es/docs/reference/rest/v2/projects.agent.environments#Environment.GenericWebService